torch/_functorch/aot_autograd.py

```diff
@@ -2443,7 +2439,8 @@ def create_aot_dispatcher_function(
         **aot_config.decompositions,
     }
 
-    log.setLevel(config.log_level)
+    if not torch._logging.is_initialized():
+        torch._dynamo.logging.init_logging()
```

Comment:
This seems extremely error prone. Are you saying that if I don't explicitly say `init_logging`, the logging won't be initialized? It seems extremely easy to forget this and then not get any logs.

In contrast, Huggingface logger automatically configures itself when you access it through any of the main configuration points e.g., https://github.com/huggingface/transformers/blob/a9bd5df16a46356463f2712dd8f6c109fa83d6f9/src/transformers/utils/logging.py#L124 I think in practice this means the logger is initialized if you import any module that uses it.

Reply:
This likely requires all of the artifacts to be registered in a single place. This is like this in order to allow init_logging to occur at the latest possible point (after all artifact types have been imported). I think if we resolve the ordering problem then this can be done at import time, and then if the user calls `set_logs`

Reply:
This is resolved by initializing when we import torch, and storing all log record types in one place.